service/src/main/kotlin/fi/espoo/evaka/invoicing/data/InvoiceQueries.kt

```diff
@@ -472,20 +474,22 @@ fun Database.Transaction.lockInvoices(ids: List<InvoiceId>) {
     createUpdate { sql("SELECT id FROM invoice WHERE id = ANY(${bind(ids)}) FOR UPDATE") }.execute()
 }
 
-fun Database.Transaction.insertInvoices(
-    invoices: List<Invoice>,
-    relatedFeeDecisions: Map<InvoiceId, List<FeeDecisionId>> = emptyMap(),
-) {
-    upsertInvoicesWithoutRows(invoices)
-    insertInvoiceRows(invoices.map { it.id to it.rows })
+fun Database.Transaction.insertDraftInvoices(
+    invoices: List<DraftInvoice>,
+    relatedFeeDecisions: Map<PersonId, List<FeeDecisionId>> = emptyMap(),
+): List<InvoiceId> {
+    val invoiceIds = insertInvoicesWithoutRows(invoices)
```

Comment:
to be safe we might want to assert that `invoices.size = invoiceIds.size` or otherwise `zip` can produce unexpected results

Reply:
👍